### PR TITLE
Added maxSuggestions functionality

### PR DESCRIPTION
**Pull request by destinmoulton**
maxSuggestions added to options definition
getSuggestionsLocal function was modified:
1. a check to tell if maxSuggestions is valid and active
2. a simple if to break out of the suggestion generator loop if the maxSuggestions has been reached

### CODE REVIEW

File: jquery.autocomplete.js [hunk 1]
```diff
@@ -254,7 +255,18 @@
       var ret = { suggestions: [], data: [] }
       q       = q.toLowerCase()
 
+      var checkMaxSuggestions = false
+      if (typeof(this.options.maxSuggestions) === 'number' &&
```
Comment (lloydwatkin) [inline jquery.autocomplete.js]: Can you surround conditionals with parenthesis please?

File: jquery.autocomplete.js [hunk 2]
```diff
@@ -254,7 +255,18 @@
...
+          (this.options.maxSuggestions%1) === 0 &&
+          this.options.maxSuggestions > 0) {
+        checkMaxSuggestions = true
+      }
+
       for (var index in arr) {
+        if ((true === checkMaxSuggestions) && (ret.suggestions.length === this.options.maxSuggestions)){
```
Comment (lloydwatkin) [inline jquery.autocomplete.js]: Space after closing ')'

(sorry I need to add linting to this project!)

**Comment by destinmoulton** [comment]
The first commit (3413d3f) adds the alterations to the parens and the additional space for the conditional readability. The second commit (46a68bb) removes a superfluous change to test.html.

**Comment by lloydwatkin** [comment]
Thanks

**Pull request merged**